libmscore/textbase.cpp

```diff
@@ -519,8 +519,73 @@ void TextFragment::draw(QPainter* p, const TextBase* t) const
       {
       QFont f(font(t));
       f.setPointSizeF(f.pointSizeF() * MScore::pixelRatio);
+#ifndef Q_OS_MACOS
+      if (f.bold()) {
+            // workaround for https://musescore.org/en/node/284218
+            // and https://musescore.org/en/node/281601
+            // only needed for certain artificially emboldened fonts
+            // see https://musescore.org/en/node/281601#comment-900261
+            // in Qt 5.12.x this workaround should be no more necessary if
+            // env variable QT_MAX_CACHED_GLYPH_SIZE is set to 1
+            p->save();
+            int deviceLogicalDpi = p->device()->logicalDpiX();
+            QScreen* screen = QGuiApplication::primaryScreen();
+            qreal screenDpi = screen->logicalDotsPerInch();
+            qreal mm = p->worldTransform().m11();
+            qreal dx = p->worldTransform().dx();
+            qreal dy = p->worldTransform().dy();
+            qreal factor = 1 / mm; // correction factor for bold text drawing
+            p->setMatrix(QMatrix(mm * factor, 0.0, 0.0, mm * factor, dx, dy));
```

Comment:
Isn’t `mm * factor` = 1 ?

Reply:
due to floating point arithmetic not exactly. But should be...

Reply:
Yes. It is a leftover because at the beginning I was manually adjusting the factor to see when the problem was disappearing.
I found that the problem is indeed the fact that the diagonal elements of the painter matrix are not 1. If you look at exported output in the current MuseScore version, you will see that for svg there is no problem (m11 is forced to 1); for pdf and png when the resolution is 360 or larger there is no problem (m11 is greater or equal to 1, since it is resolution / internal resolution and the internal resolution at the moment is 360).
Well, for pdf the synthetically emboldened fonts are not emboldened, so there is not actually a problem (if we can say so) for m11<1.
For png export, when the resolution is lower than 360, for some font dimensions we see the problem.

Reply:
I was wondering if maybe use this workaround only when painting to the screen (I also don't know what happens when using two screens at different resolutions), and let the export function unchanged. What do you think?

Reply:
MuseScore doesn't deal too well with 2 screens with different DPIs already (it always only takes the settings of the primary screen as far as I can tell), I don't expect this PR to change that